src/double.ts

```diff
@@ -32,6 +33,34 @@ export class Double extends BSONValue {
     this.value = +value;
   }
 
+  /**
+   * Attempt to create an double type from string.
+   *
+   * This method will throw a BSONError on any string input that is not representable as a IEEE-754 64-bit double.
+   * Notably, this method will also throw on the following string formats:
+   * - Strings in non-decimal formats (exponent notation, binary, hex, or octal digits)
+   * - Strings with characters other than numeric, floating point, or leading sign characters (Note: 'Infinity', '-Infinity', and 'NaN' input strings are still allowed)
+   * - Strings with leading and/or trailing whitespace
+   *
+   * Strings with leading zeros, however, are also allowed
+   *
+   * @param value - the string we want to represent as an double.
+   */
+  static fromString(value: string): Double {
```

Comment:
same comment here as on the `Int32.fromString()` PR: can we throw descriptive errors for each case we're checking for in the `if` statement?

Reply:
Yep, on it